dotcom-rendering/src/web/components/Island.tsx

```diff
@@ -1,40 +1,61 @@
 import { Placeholder } from './Placeholder';
 
-interface HydrateProps {
-	deferUntil?: 'idle' | 'visible';
+type ClientOnlyProps = {
```

Comment:
This file is mostly a drive-by refactor on the props given to `Island` components - namely to separate out each of the different `deferUntil` properties and then separately to have different sets of props for when an Island is client only or not.